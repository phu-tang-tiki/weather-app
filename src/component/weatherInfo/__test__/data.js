export default {
  api_calls: {
    queryWeather: {
      data: {
        consolidated_weather: [
          {
            id: 5055458147565568,
            weather_state_name: 'Heavy Cloud',
            weather_state_abbr: 'hc',
            wind_direction_compass: 'SW',
            created: '2021-03-28T09:31:03.501081Z',
            applicable_date: '2021-03-28',
            min_temp: 8.285,
            max_temp: 13.15,
            the_temp: 11.915,
            wind_speed: 11.45839417701954,
            wind_direction: 224.33332957252898,
            air_pressure: 1022.5,
            humidity: 75,
            visibility: 10.424123120973515,
            predictability: 71,
          },
          {
            id: 5754551886938112,
            weather_state_name: 'Clear',
            weather_state_abbr: 'c',
            wind_direction_compass: 'SW',
            created: '2021-03-28T09:31:04.291788Z',
            applicable_date: '2021-03-29',
            min_temp: 8.33,
            max_temp: 18.509999999999998,
            the_temp: 16.939999999999998,
            wind_speed: 8.710184381147432,
            wind_direction: 224.3334650015942,
            air_pressure: 1027,
            humidity: 52,
            visibility: 12.047766046289668,
            predictability: 68,
          },
          {
            id: 4717326680195072,
            weather_state_name: 'Light Cloud',
            weather_state_abbr: 'lc',
            wind_direction_compass: 'S',
            created: '2021-03-28T09:31:03.400063Z',
            applicable_date: '2021-03-30',
            min_temp: 7.475,
            max_temp: 21.259999999999998,
            the_temp: 20.045,
            wind_speed: 4.715788616728969,
            wind_direction: 187.66449361619107,
            air_pressure: 1028,
            humidity: 50,
            visibility: 13.488725841088046,
            predictability: 70,
          },
          {
            id: 5051492550574080,
            weather_state_name: 'Heavy Cloud',
            weather_state_abbr: 'hc',
            wind_direction_compass: 'SW',
            created: '2021-03-28T09:31:04.422679Z',
            applicable_date: '2021-03-31',
            min_temp: 9.135,
            max_temp: 19.715,
            the_temp: 20.44,
            wind_speed: 4.460563661984676,
            wind_direction: 216.0736552847756,
            air_pressure: 1020.5,
            humidity: 59,
            visibility: 12.88413167104112,
            predictability: 71,
          },
          {
            id: 6458867434127360,
            weather_state_name: 'Showers',
            weather_state_abbr: 's',
            wind_direction_compass: 'NE',
            created: '2021-03-28T09:31:03.717131Z',
            applicable_date: '2021-04-01',
            min_temp: 6.22,
            max_temp: 13.395,
            the_temp: 13.335,
            wind_speed: 9.95355587563297,
            wind_direction: 53.46925589065744,
            air_pressure: 1026,
            humidity: 61,
            visibility: 12.758614690209178,
            predictability: 73,
          },
          {
            id: 5639633191829504,
            weather_state_name: 'Heavy Cloud',
            weather_state_abbr: 'hc',
            wind_direction_compass: 'NNE',
            created: '2021-03-28T09:31:05.797614Z',
            applicable_date: '2021-04-02',
            min_temp: 4.04,
            max_temp: 12.434999999999999,
            the_temp: 8.93,
            wind_speed: 7.761686152867256,
            wind_direction: 27.5,
            air_pressure: 1028,
            humidity: 59,
            visibility: 9.999726596675416,
            predictability: 71,
          },
        ],
        time: '2021-03-28T12:00:07.488166+01:00',
        sun_rise: '2021-03-28T06:44:40.444291+01:00',
        sun_set: '2021-03-28T19:27:23.724662+01:00',
        timezone_name: 'LMT',
        parent: {
          title: 'England',
          location_type: 'Region / State / Province',
          woeid: 24554868,
          latt_long: '52.883560,-1.974060',
        },
        sources: [
          {
            title: 'BBC',
            slug: 'bbc',
            url: 'http://www.bbc.co.uk/weather/',
            crawl_rate: 360,
          },
          {
            title: 'Forecast.io',
            slug: 'forecast-io',
            url: 'http://forecast.io/',
            crawl_rate: 480,
          },
          {
            title: 'HAMweather',
            slug: 'hamweather',
            url: 'http://www.hamweather.com/',
            crawl_rate: 360,
          },
          {
            title: 'Met Office',
            slug: 'met-office',
            url: 'http://www.metoffice.gov.uk/',
            crawl_rate: 180,
          },
          {
            title: 'OpenWeatherMap',
            slug: 'openweathermap',
            url: 'http://openweathermap.org/',
            crawl_rate: 360,
          },
          {
            title: 'Weather Underground',
            slug: 'wunderground',
            url: 'https://www.wunderground.com/?apiref=fc30dc3cd224e19b',
            crawl_rate: 720,
          },
          {
            title: 'World Weather Online',
            slug: 'world-weather-online',
            url: 'http://www.worldweatheronline.com/',
            crawl_rate: 360,
          },
        ],
        title: 'London',
        location_type: 'City',
        woeid: 44418,
        latt_long: '51.506321,-0.12714',
        timezone: 'Europe/London',
      },
    },
  },
};
