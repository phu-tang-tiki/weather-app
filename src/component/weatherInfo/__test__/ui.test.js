import WeatherInfo from '../index';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import data from './data';

describe('citi select effect', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  let store;
  store = mockStore(data);
  describe('Citi select index', () => {
    it('renders correctly', () => {
      const { asFragment } = render(
        <Provider store={store}>
          <WeatherInfo />
        </Provider>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
