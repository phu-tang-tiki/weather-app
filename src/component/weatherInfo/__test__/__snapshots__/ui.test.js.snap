// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`citi select effect Citi select index renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-row"
    style="margin: -8px -8px -8px -8px;"
  >
    <div
      class="ant-col ant-col-4"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-card ant-card-bordered ant-card-hoverable"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="Heavy Cloud"
            src="https://www.metaweather.com/static/img/weather/hc.svg"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                Sunday
              </div>
              <div
                class="ant-card-meta-description"
              >
                <p>
                  Min: 8.29
                </p>
                <p>
                  Max: 13.15
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-4"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-card ant-card-bordered ant-card-hoverable"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="Clear"
            src="https://www.metaweather.com/static/img/weather/c.svg"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                Monday
              </div>
              <div
                class="ant-card-meta-description"
              >
                <p>
                  Min: 8.33
                </p>
                <p>
                  Max: 18.51
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-4"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-card ant-card-bordered ant-card-hoverable"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="Light Cloud"
            src="https://www.metaweather.com/static/img/weather/lc.svg"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                Tuesday
              </div>
              <div
                class="ant-card-meta-description"
              >
                <p>
                  Min: 7.47
                </p>
                <p>
                  Max: 21.26
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-4"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-card ant-card-bordered ant-card-hoverable"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="Heavy Cloud"
            src="https://www.metaweather.com/static/img/weather/hc.svg"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                Wednesday
              </div>
              <div
                class="ant-card-meta-description"
              >
                <p>
                  Min: 9.13
                </p>
                <p>
                  Max: 19.71
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-4"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-card ant-card-bordered ant-card-hoverable"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="Showers"
            src="https://www.metaweather.com/static/img/weather/s.svg"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                Thursday
              </div>
              <div
                class="ant-card-meta-description"
              >
                <p>
                  Min: 6.22
                </p>
                <p>
                  Max: 13.39
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-4"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-card ant-card-bordered ant-card-hoverable"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="Heavy Cloud"
            src="https://www.metaweather.com/static/img/weather/hc.svg"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                Friday
              </div>
              <div
                class="ant-card-meta-description"
              >
                <p>
                  Min: 4.04
                </p>
                <p>
                  Max: 12.43
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
